vantage6/server/resource/role.py

```diff
@@ -227,6 +227,10 @@ def delete(self, id):
             return {"msg": f"Role with id={id} not found."}, \
                 HTTPStatus.NOT_FOUND
 
+        if role in g.user.roles:
```

Comment:
A user *is* allowed to delete pre-defined roles. So this is not needed.  A user should not be able to remove a role from its profile.

However.. I dont think this endpoint is complete as associations to the role might still be present (e.g. users, rules)

Reply:
I do understand that the user is still able to shrink his permission space in this way. But doing it this way it is more explicit that you are removing permissions.

Reply:
Agree that they are being explicit, reversing this change